### DISCUSSION TITLE
How do you extend the strapi-admin package backend code?

### DISCUSSION THREAD

**Discussion by gtamas** [strapi/strapi#6365] [Legacy Topics]
I know there is an article about this, but that covers only the react FE (admin/src/*).
Could you please tell me how do you extend the backend parts?

I have /admin/services/auth.js. 
I also have /admin/controllers/Auth.js.

These files exist in the strapi-admin package, but running "yarn develop --watch-admin" command doesn't seem to copy them, nor does it seem to detect the changes. The admin panel seems to use the original files from node_modules/strapi-admin package, no matter what I do.

How do I extend these backend files?

Thanks in advance!

**Comment by gtamas** [comment]
@lauriejim 
...Anyone? 
The usual mechanism doesn't seem to work here. Is this a big or missing feature?
I think strapi-admin should be extensible the same way you extend plugins.

Do I have to use custom webpack config to implement this?

**Comment by gtamas** [comment]
Your have fork Strapi for this, since the admin package is not a plugin yet.
Then you can modify the strapi-admin package and install the dependency directly from your fork.

It's a monorepo, so this will be useful:

https://gitpkg.now.sh/

**Comment by lauriejim** [comment]
Yes sorry for the delay.
There is 4 things that you can't customise by default without doing a fork.
- the backend of the admin
- the strapi-utils package
- the strapi (core) package
- database connectors

But a thing you can do is not fork Strapi entirely, just create a new folder in your app,
copy the package you want to customize,
and in you package.json file, use the file pointer to set the new deps location.

could be a way to customize one of these 4 points.
